src/test/java/com/work/management/EmployeeServiceTest.java

```diff
@@ -0,0 +1,56 @@
+package com.work.management;
+
+import static org.junit.Assert.assertEquals;
+import static org.mockito.Mockito.doReturn;
+
+import com.work.management.dto.EmployeeDto;
+import com.work.management.entity.Employee;
+import com.work.management.repository.EmployeeRepository;
+import com.work.management.service.employee.impl.EmployeeServiceImpl;
+import java.util.Optional;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+import org.springframework.beans.BeanUtils;
+import org.springframework.boot.test.context.SpringBootTest;
+
+@RunWith(MockitoJUnitRunner.class)
+@SpringBootTest
+public class EmployeeServiceTest {
+
+  @InjectMocks
+  private EmployeeServiceImpl employeeService;
+
+  @Mock
+  private EmployeeRepository employeeRepository;
+
```

Comment:
Use in memory db to test code instead of mocking repositories. https://www.baeldung.com/spring-jpa-test-in-memory-database